lib/adapters/http/read.js

```diff
@@ -1,13 +1,94 @@
+'use strict';
 var _ = require('underscore');
 var contentType = require('content-type');
-var Juttle = require('../../runtime').Juttle;
+var AdapterRead = require('../../runtime/adapter-read');
 var parsers = require('../parsers');
+var errors = require('../../errors');
 var request = require('request');
+var Promise = require('bluebird');
+
+class Iterator {
+    constructor(adapter, req, res) {
+        this.points = [];
+        this.done = false;
+
+        var format;
+        try {
+            format = contentType.parse(res).type.split('/')[1];
```

Comment:
Yeah I agree -- it turns out that either works, but it does seem confusing to parse the content type of the request and not the response.

I tightened things up so we don't use the `req` object at all any more.